django-stubs/views/decorators/__init__.pyi

```diff
@@ -0,0 +1,13 @@
+from typing import Any, Protocol, TypeVar
+
+from django.http.request import HttpRequest
+from django.http.response import HttpResponseBase
+
+# `*args: Any, **kwargs: Any` means any extra argument(s) can be provided, or none.
+class _View(Protocol):
+    def __call__(self, request: HttpRequest, /, *args: Any, **kwargs: Any) -> HttpResponseBase: ...
+
+class _AsyncView(Protocol):
+    async def __call__(self, request: HttpRequest, /, *args: Any, **kwargs: Any) -> HttpResponseBase: ...
+
+_ViewFuncT = TypeVar("_ViewFuncT", bound=_View | _AsyncView)  # noqa: PYI018
```

Comment:
btw `django.db.transaction.non_atomic_requests` decorator expects views as well, and could re-use this.